File: api/krameriusplus-core/src/main/java/cz/inqool/dl4dh/krameriusplus/core/system/job/jobevent/JobStatus.java
package cz.inqool.dl4dh.krameriusplus.core.system.job.jobevent;

import org.springframework.batch.core.BatchStatus;

public enum JobStatus {
    CREATED,
    COMPLETED,
    STARTING,
    STARTED,
    STOPPING,
    STOPPED,
    FAILED,
    ABANDONED,
    UNKNOWN;

    public static JobStatus from(BatchStatus status) {
        return JobStatus.valueOf(status.name());
    }
}

File: web/src/enums/job-status.ts
/**
 * DO NOT USE this enum to deserialize values from BE, not all values are in this enum. Use only
 * for filtering options
 */
export enum JobStatus {
	CREATED = 'CREATED',
	COMPLETED = 'COMPLETED',
	STARTED = 'STARTED',
	STOPPED = 'STOPPED',
	FAILED = 'FAILED',
}
